tools/CouchDB_MRI_Importer.php

```diff
@@ -336,9 +336,10 @@ public function getScanTypes()
     {
 
         $ScanTypes = $this->SQLDB->pselect(
-            "SELECT DISTINCT msc.Scan_type as ScanType from mri_scan_type msc
-JOIN files f ON msc.ID= f.AcquisitionProtocolID
-JOIN files_qcstatus fqc ON f.FileID=fqc.FileID ORDER BY f.AcquisitionProtocolID",
+            "SELECT DISTINCT msc.Scan_type as ScanType, f.AcquisitionProtocolID from mri_scan_type msc
+            JOIN files f ON msc.ID= f.AcquisitionProtocolID
+            JOIN files_qcstatus fqc ON f.FileID=fqc.FileID
+            ORDER BY f.AcquisitionProtocolID",
```

Comment:
I must be missing something, because I don't see what's different here other than the whitespace?

Reply:
@driusan the SELECT statement now has `f.AcquisitionProtocolID` for mysql 5.7 to not throw and error

which makes sens cause we are grouping by `f.AcquisitionProtocolID`

Reply:
Oh, I missed that.. but there's no `GROUP BY` AcquisitionProtocolID, it's an `ORDER BY`, so I still don't see why it should matter? You can ORDER BY something that isn't selected.

In fact, I just tried the old query on MySQL 5.7 and it seems to be working fine for me. There must be something else going on

```
mysql> SELECT DISTINCT msc.Scan_type as ScanType from mri_scan_type msc JOIN files f ON msc.ID= f.AcquisitionProtocolID  JOIN files_qcstatus fqc ON f.FileID=fqc.FileID ORDER BY f.AcquisitionProtocolID;
+----------------------+
| ScanType             |
+----------------------+
| t1w                  |
| t2w                  |
| dti                  |
| ep2d_bold            |
| DTI65                |
| DTI65_B1000          |
| DTIPrepReg           |
| DTIPrepNoReg         |
| DTIPrepQCReport      |
| DTIPrepXMLQCReport   |
| DTIPrepDTIColorFA    |
| DTIPrepDTIFA         |
| DTIPrepDTIMD         |
| DTIPrepBaseline      |
| DTIPrepIDWIBetMask   |
| DTIPrepDTI           |
| DTIPrepIDWI          |
| DTIPrepVCXMLQCResult |
| DTIPrepVC            |
| DTIPrepVCDTI         |
| DTIPrepVCBaseline    |
| DTIPrepVCIDWIBetMask |
| DTIPrepVCIDWI        |
| DTIPrepVCDTIFA       |
| DTIPrepVCDTIcolorFA  |
| DTIPrepVCDTIMD       |
+----------------------+
26 rows in set (0.21 sec)

mysql> SELECT VERSION()
    -> ;
+-----------+
| VERSION() |
+-----------+
| 5.7.9     |
+-----------+
1 row in set (0.00 sec)
```

Reply:
@driusan can you print your sql_mode

Reply:
Okay, I've managed to reproduce it. It's because of the DISTINCT that it doesn't know how to `ORDER BY` something that's not selected.

```
mysql> select @@sql_mode;
+--------------------------------------------+
| @@sql_mode                                 |
+--------------------------------------------+
| STRICT_TRANS_TABLES,NO_ENGINE_SUBSTITUTION |
+--------------------------------------------+
1 row in set (0.00 sec)

mysql> SET sql_mode='ONLY_FULL_GROUP_BY';                                                                                                                           Query OK, 0 rows affected (0.00 sec)

mysql> SELECT DISTINCT msc.Scan_type as ScanType from mri_scan_type msc JOIN files f ON msc.ID= f.AcquisitionProtocolID  JOIN files_qcstatus fqc ON f.FileID=fqc.FileID ORDER BY f.AcquisitionProtocolID;
ERROR 3065 (HY000): Expression #1 of ORDER BY clause is not in SELECT list, references column 'ibis.f.AcquisitionProtocolID' which is not in SELECT list; this is incompatible with DISTINCT
```
That makes more sense.